orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -273,21 +277,42 @@ func (m *swiftDialogMDMMigrator) renderError() (chan swiftDialogExitCode, chan e
 // unenroll, an error is returned.
 func (m *swiftDialogMDMMigrator) waitForUnenrollment() error {
 	maxRetries := int(mdmUnenrollmentTotalWaitTime.Seconds() / m.unenrollmentRetryInterval.Seconds())
-	fn := m.testEnrollmentCheckFn
-	if fn == nil {
-		fn = func() (bool, error) {
+	checkFileFn := m.testEnrollmentCheckFileFn
+	if checkFileFn == nil {
+		checkFileFn = func() (bool, error) {
 			return file.Exists(mdmEnrollmentFile)
 		}
 	}
+	checkStatusFn := m.testEnrollmentCheckStatusFn
+	if checkStatusFn == nil {
+		checkStatusFn = func() (bool, string, error) {
+			return profiles.IsEnrolledInMDM()
+		}
```

Comment:
nit: in this case we can do:

```suggestion
		checkStatusFn = profiles.IsEnrolledInMDM
```